### ISSUE TITLE
The .babelrc is ignored during transpilation for styled-components

### ISSUE THREAD

**Issue by shlajin** [martpie/next-transpile-modules#116]
- [X] I HAVE READ [THE FAQ](https://github.com/martpie/next-transpile-modules#faq) AND MY PROBLEM WAS NOT DESCRIBED THERE
- [X] I WILL GIVE 10$ TO CHARITY IF MY SOLUTION WAS ACTUALLY IN THE FAQ

**Are you trying to transpile a local package or an npm package?**
A local package (monorepo, yarn workspaces, sibling package)

**Describe the bug**
I have this in my .babelrc
```
{
  "presets": [
    "next/babel"
  ],
  "plugins": [
    [
      "styled-components",
      {
        "ssr": true
      }
    ]
  ]
}
```

When I transpile the current (nextjs repo) package, I get nice class names with styled-components. E.g.
```
const Test = styled.h1`
  color: red;
`
...
<Test>Nice</Test>
```

```
<h1 class="pages__Test-sc-1l62dhw-0 ehBOK">Nice</h1>
```

I look at the class name and see that the component name is `Test`, and it's located somewhere in `pages`.

In my sibling package (named `dashboard`), I do the same thing:
```
export const NotWorkingTest = styled.h1`
  color: pink;
`
```

then in my next.js repo:
```
// relative or absolute import - doesn't matter
import { NotWorkingTest } from "dashboard/src/..."
...
<NotWorkingTest>Not very nice</NotWorkingTest>
```

... and styles don't apply. It spawns a error in browser's console, though:

```
Prop `className` did not match. Server: "sc-daBvwG enrxww" Client: "sc-ksluoS cAoCyc"
```

this **must** be something related to `ssr: true` in my `.babelrc` file. If I restart the server, I the correct pink color of my second h1, but only until I hit refresh. After, same `className did not match` error. This bug sounds huge, but it's hard to reproduce and does not always trigger, I'm still investigating. Maybe something to do with my next.config.js.

What happens always (or, I'd say, doesn't happen) is the class name exposes some debug info. Instead, the class name is just random characters, e.g. `sc-ksluoS cAoCyc`. Different combinations of `ssr`, `displayName` and `preprocess` on styled-component's .babelrc have no effect on the class name.

**To Reproduce**
I put it in the description of the bug, but please let me know if you need more info

**Expected behavior**
I see nice classnames when I transpile sibling packages without any errors in the console

**Setup**

- Next.js version: 9.5.3
- `next-transpile-modules` version: 4.1.0
- Node.js version: 13.9.0
- `npm`/`yarn` version: yarn 1.22.0
- Operating System: Mac OS X
- Webpack 4 or 5: 4

**Additional context**
Possibly the same issue: https://github.com/martpie/next-transpile-modules/issues/99

**Comment by martpie** [comment]
Be ready to give 10$ to charity I'm afraid. Try with `babel.config.js` and not `.babelrc`

**Comment by shlajin** [comment]
Ok, deal is deal, 10$ goes to one local charity that helps with accommodation for elderly people!

Although you stated almost under every issue that the problem is `.babelrc`, mine wasn't there and I did not I realize I should try it (and I tried almost everything else in the world lol). Is there any legit cases when `.babelrc` is required and `babel.config.js` is not when used together with this library? I'm thinking that maybe it's a good idea to put in bold "If you're using .babelrc, don't"

Thank you for your help and your work, very appreciate it!
